src/cpu/flexcpu/simple_dataflow_thread.cc

```diff
@@ -757,6 +870,22 @@ SDCPUThread::populateDependencies(shared_ptr<InflightInst> inst_ptr)
 
     // END ISA explicit serialization
 
+    if (static_inst->isNonSpeculative()) {
+        const shared_ptr<InflightInst> last_inst = *(++inflightInsts.rbegin());
+
+
+        if (!(last_inst->isCommitted() || last_inst->isSquashed()))
```

Comment:
(not in this patch) We should factor this out into a named function (e.g., `instCompleted(last_inst)`). I'm not sure what the right name is :). I've just seen this condition written multiple times, so it makes me think it should be a function.

Reply:
I mean this check is really a "should never be the case" check, since in theory, by the time I'm adding a new instruction, committed instructions should have been taken off the buffer, and nothing should follow a squashed instruction at this time :T. I just need to spend time to see if those assertions hold true before I remove those barriers. I think I ran into problems early on with unresolvable dependencies due to things being in the same call stack though...